### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by raymens** [Horusiath/Akkling#11]
I see some great enhancements added as issues, could you add a roadmap on what issues you (or others) are working on?

**Comment by Horusiath** [comment]
Right now I'm working on them alone and in irregular manner, so no explicit roadmap was necessary. But  it's acutally a good idea, no matter if we speak about single person or collaboration development.

**Comment by raymens** [comment]
Yeah it would be great to see what you (or anyone) are actually working on and which of them are just idea floating around for now. It's also easier for any potential contributors to know what they can grab for a PR.

**Comment by Horusiath** [comment]
While this will need to wait for a proper documentation, initial roadmap looks like this:

### 0.3 

- Publish Akkling.TestKit to make F#-idiomatic API over Akka.TestKit
- (Hopefully) Implement full hot code loading on `spawne` - for now it requires to define all of the scope to be loaded in the single quotation. The goal is to span it also over dependent components used but not defined inside quotation.
- tbc...

### 0.2

- Simplified existing Akka.FSharp API to necessary minimum.
- Introduced effects - this way actor can return special kind of effect to trigger some behavior on actor like `Unhandled`, `Stop` or `Ignore`.
- Introduced actor's lifecycle events: a way to hook to `PreStart`, `PostStop`, `PreRestart`, `PostRestart` actions through dedicated messages. This also replaces `mailbox.Defer` callbacks.
- Created whole new API for Akkling.Persistence: now it works as single uniform function, which is invoked on both OnCommand, OnRecover, and on persist and deffer callbacks. Those phases can be distinguished using correlated active patterns.

### 0.1

- Initial release - fork from Akka.FSharp
- Implemented typed actor refs.

**Comment by Horusiath** [comment]
@gsomix right now I'm mostly occupied with implementing changes in akka.net core repository. Some of the changes I was thinking about changing how actor handles incoming messages - split this behavior in two (one for user-defined messages and one for system-defined ones). This would allow us to avoid retyping actor refs in situations when your actor needs to react on both user and system messages. Problem with that is that is a breaking change and Akkling is already used in many production systems.

**Comment by Kavignon** [comment]
Since Akkling is being used in many production environment, do you still consider this as an experimental project?

**Comment by Swoorup** [comment]
I don't really mind the breaking changes, as long as major version is changed? As is, it feels odd to use Actor<obj> to use System messages
